lib/auth/session_access.go

```diff
@@ -223,22 +227,30 @@ func (e *SessionAccessEvaluator) hasPolicies() bool {
 
 // Generate a pretty-printed string of precise requirements for session start suitable for user display.
 func (e *SessionAccessEvaluator) PrettyRequirementsList() string {
-	s := "require all:"
+	var s strings.Builder
+	s.WriteString("require all:")
 
 	for _, policySet := range e.policySets {
 		policies := e.extractApplicablePolicies(policySet)
 		if len(policies) == 0 {
 			continue
 		}
 
-		s += fmt.Sprintf("\n\t  one of (%v):", policySet.Name)
+		s.WriteString("\n\t  one of (")
+		s.WriteString(policySet.Name)
+		s.WriteString("):")
 
 		for _, require := range policies {
-			s += fmt.Sprintf("\n\t    - %vx %v with mode %v", require.Count, require.Filter, strings.Join(require.Modes, ","))
+			s.WriteString("\n\t    - ")
```

Comment:
builder implements `io.Writer`, so you can use `fmt.Fprintf(s, "", ...)` if that's cleaner.